app/src/main/java/com/android/voyageur/ui/trip/schedule/WeeklyView.kt

```diff
@@ -0,0 +1,179 @@
+                  fontWeight = FontWeight.Bold,
+                  modifier = Modifier.padding(bottom = 8.dp).testTag("weekRange_$weekIndex"))
+
+              for (dayOffset in 0..6) {
+                val currentDate = weekStart.plusDays(dayOffset.toLong())
+                val activitiesForDay =
+                    activities.filter { activity ->
+                      val activityDate =
+                          activity.startTime
+                              .toDate()
+                              .toInstant()
+                              .atZone(ZoneId.systemDefault())
+                              .toLocalDate()
+                      activityDate == currentDate
+                    }
+
+                DayActivityCount(
+                    date = currentDate,
+                    activityCount = activitiesForDay.size,
+                    navigationActions = navigationActions,
+                    trip = trip,
+                    dayIndex = dayOffset,
+                    weekIndex = weekIndex,
+                    onDaySelected = onDaySelected // Pass the callback
+                    )
+              }
+            }
+      }
+}
+
+@Composable
+private fun DayActivityCount(
+    date: LocalDate,
+    activityCount: Int,
+    navigationActions: NavigationActions,
+    trip: Trip,
+    dayIndex: Int,
+    weekIndex: Int,
+    onDaySelected: () -> Unit // Add this parameter
+) {
+  Button(
+      onClick = onDaySelected, // Use the callback
+      modifier = Modifier.fillMaxWidth().testTag("dayButton_${weekIndex}_$dayIndex"),
+      shape = RoundedCornerShape(24.dp)) {
+        Row(
+            modifier = Modifier.padding(12.dp),
+            horizontalArrangement = Arrangement.spacedBy(4.dp)) {
+              Text(
+                  text = "${date.dayOfWeek.toString().take(1)} ${date.dayOfMonth}",
+                  style = MaterialTheme.typography.bodyLarge,
+                  fontWeight = FontWeight.Medium,
+                  modifier = Modifier.testTag("dayText_${weekIndex}_$dayIndex"))
+
+              Text(
+                  text = "  -  $activityCount activities",
+                  style = MaterialTheme.typography.bodyLarge,
+                  modifier = Modifier.testTag("activityCount_${weekIndex}_$dayIndex"))
+            }
+      }
+}
+
+fun generateWeeks(startTimestamp: Timestamp, endTimestamp: Timestamp): List<DateRange> {
+  val startDate = startTimestamp.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+  val endDate = endTimestamp.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+
+  val weeks = mutableListOf<DateRange>()
+
+  var currentWeekStart = startDate.with(DayOfWeek.MONDAY)
+  val lastWeekStart = endDate.with(DayOfWeek.MONDAY)
+
+  while (currentWeekStart <= lastWeekStart) {
+    weeks.add(DateRange(first = currentWeekStart, last = currentWeekStart.plusDays(6)))
+    currentWeekStart = currentWeekStart.plusWeeks(1)
+  }
+
+  return weeks
+}
```

Comment:
It works fine, but it also includes some dates before the start of the trip if the trip does not start on a Monday. It might be worth taking a look as a possible improvement.

Reply:
I agree, maybe the big card can remain in this case Nov 4 - Nov 10, but inside the card we could have only the days which are part of the trip

Reply:
Thank you for your remarks! I agree with you, I changed the code and fixed this issue.